parser/pbf/pbf.go

```diff
@@ -30,20 +34,47 @@ func readDenseNodes(
 	coordScale := 0.000000001
 	lastKeyValPos := 0
 
+	var lastDenseinfoTimestamp int64
+	var lastDenseinfoChangeset int64
+	var lastDenseinfoUid int32
+	var lastDenseinfoUserSid int32
+
 	for i := range coords {
 		lastId += dense.Id[i]
 		lastLon += dense.Lon[i]
 		lastLat += dense.Lat[i]
+
+		lastDenseinfoTimestamp += dense.Denseinfo.Timestamp[i]
+		lastDenseinfoChangeset += dense.Denseinfo.Changeset[i]
+		lastDenseinfoUid += dense.Denseinfo.Uid[i]
+		lastDenseinfoUserSid += dense.Denseinfo.UserSid[i]
+
 		coords[i].Id = lastId
 		coords[i].Long = (coordScale * float64(lonOffset+(granularity*lastLon)))
 		coords[i].Lat = (coordScale * float64(latOffset+(granularity*lastLat)))
+
+		DenseinfoVersion := dense.Denseinfo.Version[i]
+
+		DenseinfoTimestamp := time.Unix(lastDenseinfoTimestamp, 0)
+		DenseinfoChangeset := lastDenseinfoChangeset
+		DenseinfoUid := lastDenseinfoUid
+		DenseinfoUserSid := lastDenseinfoUserSid
+
 		if stringtable != nil && len(dense.KeysVals) > 0 {
 			if dense.KeysVals[lastKeyValPos] != 0 {
 				tags := parseDenseNodeTags(stringtable, &dense.KeysVals, &lastKeyValPos)
+
 				if tags != nil {
+
 					if _, ok := tags["created_by"]; ok && len(tags) == 1 {
 						// don't add nodes with only created_by tag to nodes cache
 					} else {
+						tags["osm_version"] = strconv.FormatInt(int64(DenseinfoVersion), 10)
```

Comment:
Can you add a parseMetadata option to the read functions and only add the tags if it is set? We can see later how we make this option available.